src/wiseagents/core.py

```diff
@@ -692,14 +692,14 @@ def system_message(self) -> Optional[str]:
         return self._system_message
 
     @property
-    def pre_user_message(self) -> Optional[str]:
-        """Get the pre user message associated with the agent."""
-        return self._pre_user_message
+    def pre_user_messages(self) -> Optional[List[str]]:
+        """Get the list of pre user messages associated with the agent."""
+        return self._pre_user_messages
 
     @property
-    def post_user_message(self) -> Optional[str]:
-        """Get the post user message associated with the agent."""
-        return self._post_user_message
+    def post_user_messages(self) -> Optional[List[str]]:
```

Comment:
So the framework core simply never uses these pre and post-messages? All is up to the custom implementation. I'm fine with that for now, we may want to consider a more generic use of them in the future

Reply:
Right, currently it's up to the implementation to decide how to make use of these. Have created https://github.com/wise-agents/wise-agents/issues/393 to track this.